### PR TITLE
Display checkmark icon in PR files if file is marked as viewed

### LINKED ISSUE #98: Manage file viewed state in PR
Hello,

Once again, thanks for the awesome plugin.
I would love a feature to handle file viewed states in PRs, which is really usefull for huge PRs that get updated between reviews.

Github provides the feature in its UI top right of diffs here and it gets unchecked if the file changes after the review

image

I could not find stuff about it in the rest API from github, but I saw the `viewerViewedState` attribute in the graphql `PullRequestChangedFile` object: https://docs.github.com/en/graphql/reference/objects#pullrequestchangedfile

We could maybe build something around it ?

I think we can decompose this in two different parts

- [x] Add a visual feedback of the viewed state in the PR status tree next to the file names
- [x] Provide a way to manipulate this viewed state (toggle, viewed, and not viewed)

If it seems sensible I will look into it, not very familiar with graqhql apis though but they seem quite cool

### PR DESCRIPTION
**Pull request by GuillaumeLagrange**
This pr queries the [pull request file viewed state](https://docs.github.com/en/graphql/reference/objects#pullrequestchangedfile) from github's graphql api, and displays a small check mark is the file has been marked as viewed.

image

Resolves #98

### CODE REVIEW

**Comment by GuillaumeLagrange** [comment]
@ldelossa this is ready for review, I added the toggle command :smile:

File: lua/litee/gh/pr/state.lua [hunk 1]
```diff
@@ -513,16 +540,18 @@ function M.load_state_async(pull_number, on_load)
...
-                    M.get_reviews_async(pull_number, M.pull_state.user["login"], function()
-                        M.get_review_threads_async(pull_number, function()
-                            M.get_commits_async(pull_number, function()
-                                M.get_check_runs(
-                                    function ()
-                                        vim.schedule(function() vim.api.nvim_echo({{spinner() .. " successfully fetched all PR state.", "LTSuccess"}}, false, {}) end)
-                                        vim.schedule(on_load)
-                                    end
-                                )
+                M.get_pull_files_viewed_state_async(pull_number, function()
```
Comment (ldelossa) [inline lua/litee/gh/pr/state.lua:543]: LOL, you are brave to embrace this monstrosity of a function.

**Review by ldelossa — APPROVED** [review]
Looks great to me

**Comment by ldelossa** [comment]
BTW @GuillaumeLagrange I'm thinking of starting a re-implementation of this plugin using nvim-ide framework:

https://github.com/ldelossa/nvim-ide

That code is way prettier and easier to maintain. If you're interested in these efforts we can discuss further.

**Comment by GuillaumeLagrange** [comment]
@ldelossa that could be interesting indeed, I will look into `nvim-ide`.

Is there some issue with the theoretical first steps to do so ? Or is it just a plan that you had that we would need to clear out ?

**Comment by ldelossa** [comment]
@GuillaumeLagrange No issue, and its just a lofty goal right now, its been floating around my head. TBH, I got burnt out from dealing with Github API and the weird way you need to dance around things. But it could be nice to start working on a nvim-ide focused implementation, one component at a time.

**Pull request merged**